internal/summon/command/run.go

```diff
@@ -37,20 +38,48 @@ func buildEnvironment(secrets map[string]string, secretsMap secretsyml.SecretsMa
 // resolveSecrets obtains the value of each requested secret.
 func resolveSecrets(provider plugin_v1.Provider, secretsMap secretsyml.SecretsMap) (result map[string]string, err error) {
 	result = make(map[string]string)
+
+	var varSecretsSpecKeys []string
+	var varSecretsSpecPaths []string
+
 	for key, spec := range secretsMap {
-		var value string
 		if spec.IsVar() {
-			var valueBytes []byte
-			if valueBytes, err = provider.GetValue(spec.Path); err != nil {
-				return
-			}
-			value = string(valueBytes)
+			varSecretsSpecKeys = append(varSecretsSpecKeys, key)
+			varSecretsSpecPaths = append(varSecretsSpecPaths, spec.Path)
 		} else {
 			// If the spec isn't a variable, use its value as-is
-			value = spec.Path
+			value := spec.Path
+			result[key] = value
+		}
+	}
+
+
+	// Get the variable values
+	providerResponses, err := provider.GetValues(varSecretsSpecPaths...)
```

Comment:
This should probably only be done if `if spec.IsVar()` is evaluated to true. You may want to consider an early exit instead when `if spec.IsVar()` is false earlier in the func.

Reply:
I don't follow. `varSecretsSpecPaths` is only populated from specs that are truthy for `spec.IsVar()`.

Reply:
You're right - I totally didn't explain well lol. What I meant is that if `varSecretsSpec*` are empty, there's no need to send any requests to the provider.